util/topgen.py

```diff
@@ -823,26 +862,52 @@ def _process_top(topcfg, args, cfg_path, out_path, pass_idx):
     # load Hjson and pass validate from reggen
     try:
         ip_objs = []
-        for x in ips:
+        for ip_desc_file in ips:
+            ip_name = ip_desc_file.stem
             # Skip if it is not in the module list
-            if x.stem not in [ip["type"] for ip in topcfg["module"]]:
+            if ip_name not in [ip["type"] for ip in topcfg["module"]]:
                 log.info("Skip module %s as it isn't in the top module list" %
-                         x.stem)
+                         ip_name)
                 continue
 
             # The auto-generated hjson might not yet exist. It will be created
             # later, see generate_{ip_name}() calls below. For the initial
-            # validation, use the template in hw/ip/{ip_name}/data .
-            if x.stem in generated_list and not x.is_file():
-                hjson_file = ip_dir / "{}/data/{}.hjson".format(x.stem, x.stem)
-                log.info(
-                    "Auto-generated hjson %s does not yet exist. " % str(x) +
-                    "Falling back to template %s for initial validation." %
-                    str(hjson_file))
+            # validation, use the Hjson file with default values.
+            # TODO: All of this is a rather ugly hack that we need to get rid
```

Comment:
@imphil sorry i'm a bit late to the party.
Does this comment mean hjson's won't be templated in the future? or that they will be templated in a specific way?

Reply:
First of all: in this PR and the first stage of ipgen work, nothing changes here from how we did it before: we use the default template in the first run, and then the real template in the second run of topgen.

In the medium-term future, I'd hope that we can avoid string-templating Hjson files. Instead, it would be great if we can use what are template variables now as proper variables in Hjson files. But all of this is for another day.

Reply:
ah yeah i think the main thing we run into there is when we using a templating variable to create different versions of a register.. we don't really support the idea of nested multi-reg right now.